app/src/split/bluetooth/central.c

```diff
@@ -328,6 +328,12 @@ static void split_central_process_connection(struct bt_conn *conn) {
 
     LOG_DBG("New connection params: Interval: %d, Latency: %d, PHY: %d", info.le.interval,
             info.le.latency, info.le.phy->rx_phy);
+
+    // keep scanning until all peripherals are connected
+    for (int i = 0; i < ZMK_BLE_SPLIT_PERIPHERAL_COUNT; i++) {
+        if (peripherals[i].state != PERIPHERAL_SLOT_STATE_CONNECTED)
+            start_scan();
```

Comment:
I think this might be fine for now, but I'm curious if we need to at all considering switching to a lower duty scanning if we fail to find a new peripheral in a certain amount of time. Constant scanning will draw more battery if we have a multi-peripheral with something like an "optional numberpad" that a user doesn't bother connecting/charging regularly.

Reply:
I agree, would this be a new `start_scan_limited()` function with a longer `bt_le_scan_param::interval` value set? 

I also had a concern regarding security. If we fail to connect one of the peripherals at power-up and we continue scanning indefinitely, then a third party with any ZMK peripheral in range will be able to hijack that peripheral slot and get up to funny business, right? Maybe a better solution is to allow manual triggering of the split pairing from a key-combination after the initial pairing times out.

Ideally I'd prefer authenticated pairing with a pre-shared key as an option, but that is a separate issue.

Reply:
The way security is handled in my initial dongle PR is that the central stores the MAC address of the peripherals and will only reconnect to those once all the devices are known.

See https://github.com/slicemk/zmk/blob/a14004531d2ff6935c2c6c8bcaabcb1c0f5c3572/app/src/ble.c#L284-L312 and https://github.com/slicemk/zmk/blob/a14004531d2ff6935c2c6c8bcaabcb1c0f5c3572/app/src/split/bluetooth/central.c#L265-L271

Reply:
This code works for me until restarting the dongle. It seems like the peripheral addresses are not stored/restored, or am I doing something wrong here? Should I include https://github.com/zmkfirmware/zmk/pull/836 as well?

Reply:
I'm fine doing the address work as a follow up. 

I do think the scanning tweaks should get in as part of this though.

Reply:
I just started testing the latest ZMK today with the address stuff and also an updated https://github.com/slicemk/zmk/blob/a14004531d2ff6935c2c6c8bcaabcb1c0f5c3572/app/src/split/bluetooth/central.c#L323-L353 for scanning. I'll probably update my old PR in the next day or two.